### DISCUSSION TITLE
NAND Gate transistor layout can display wrong value

### DISCUSSION THREAD

**Discussion by Kitick** [flandreas/antares#197] [General]
When connecting transistors up in a NAND gate configuration, and powering on A, pulls the value to 0, even though B is still outputting a value. And if you enable B then disable, it then works.

Here's some images:
![image](https://user-images.githubusercontent.com/34259991/121838473-c7880500-cc8c-11eb-9c8f-8bd6dfead9bb.png)
![image](https://user-images.githubusercontent.com/34259991/121838493-d4a4f400-cc8c-11eb-9981-94e218b261e8.png)
![image](https://user-images.githubusercontent.com/34259991/121838522-e4bcd380-cc8c-11eb-9154-466069fa55c8.png)

**Comment by flandreas** [comment]
Note that transistors as they are currently implemented in Antares are just another representation of controlled buffers. Therefore, they are not bi-directional.

I'm not an expert in transistor-based circuits, but 'd say that what you are trying to do here is RTL logic, which I don't think will work with the current implementation, right? That's also why the example project "Logic Gates from transistors" use TTL logic.

But please correct me if I'm wrong.

**Comment by richardabendroth** [comment]
I think this is what @flandreas means: 
![image](https://user-images.githubusercontent.com/25486087/121899697-f356c980-cd24-11eb-9caf-46bf10a4d080.png)

My question would be: Shouldn't the FET have a body diode?

**Comment by Kitick** [comment]
Ah, that makes sense, I've heard of RTL and TTL logic based transistors, however I'm not the best at them either.
What I think the problem is though is Antares makes the 0 flow though, and then it forgets the the other transistor is outputting a 1. And only when you re-assert B does it remember.
